src/corppa/poetry_detection/core.py

```diff
@@ -85,16 +93,31 @@ class Excerpt:
     detection_methods: set[str]
     # Optional notes field
     notes: Optional[str] = None
+    # Excerpt id, set in post initialization
+    excerpt_id: str = field(init=False)
 
     def __post_init__(self):
-        # Check that dectection method set is not empty
-        if not self.detection_methods:
-            raise ValueError("Must specify at least one detection method")
         # Check PPA span indices
         if self.ppa_span_end <= self.ppa_span_start:
             raise ValueError(
                 f"PPA span's start index {self.ppa_span_start} must be less than its end index {self.ppa_span_end}"
             )
+        # Check that dectection method set is not empty
+        if not self.detection_methods:
+            raise ValueError("Must specify at least one detection method")
+
+        # Set excerpt id
+        # TODO: Define behavior for multiple detection methods
+        if len(self.detection_methods) == 1:
+            [detect_name] = self.detection_methods
+            if detect_name not in DETECTION_PFX_TABLE:
+                raise ValueError(f"Unsupported detection method '{detect_name}'")
+            detect_pfx = DETECTION_PFX_TABLE[detect_name]
+            self.excerpt_id = f"{detect_pfx}@{self.ppa_span_start}:{self.ppa_span_end}"
+        else:
+            # TODO: Should all detection methods be validated?
+            # TODO: Is this the tag we want for multiple detection methods?
+            self.excerpt_id = f"mult@{self.ppa_span_start}:{self.ppa_span_end}"
```

Comment:
For the detection method check, it's more pythonic to just try to access and then catch if it's not found.

I suggest adjusting the logic so the id construction is done once.

```suggestion
        if len(self.detection_methods) == 1:
            [detect_name] = self.detection_methods
            try:
                id_prefix = DETECTION_PFX_TABLE[detect_name]
            except KeyError:
                raise ValueError(f"Unsupported detection method '{detect_name}'")
            
            self.excerpt_id = f"{detect_pfx}@{self.ppa_span_start}:{self.ppa_span_end}"
        else:
            id_prefix = "m" 
           
        self.excerpt_id = f"{id_prefix}@{self.ppa_span_start}:{self.ppa_span_end}"        
```

Reply:
Regarding your TODOs:
- yes, probably should validate detection methods but it doesn't need to be implemented now, it's not what you're working on (you can leave a todo if you want)
- can we reserve `m` for the prefix of combined records? m for multi/merge

Reply:
ooh but when we do the detection method validation we should use set logic

Reply:
No, multi/merge cannot be m because "m" is already in use for "manual"

Reply:
ooh, c for combined / combination is a nice alternative.

maybe there's a better alternative to manual; but I wasn't sure. it's a catch-all for non-adjudication manual entries, but maybe we can make a quick decision tomorrow morning.

Reply:
Setting to "c" for combined.